plugins/inputs/vsphere/endpoint.go

```diff
@@ -653,6 +672,35 @@ func getClusters(ctx context.Context, e *Endpoint, resourceFilter *ResourceFilte
 	return m, nil
 }
 
+//noinspection GoUnusedParameter
+func getResourcePools(ctx context.Context, e *Endpoint, resourceFilter *ResourceFilter) (objectMap, error) {
+	var resources []mo.ResourcePool
+	err := resourceFilter.FindAll(ctx, &resources)
+	if err != nil {
+		return nil, err
+	}
+	m := make(objectMap)
+	for _, r := range resources {
+		m[r.ExtensibleManagedObject.Reference().Value] = &objectRef{
+			name:         r.Name,
+			ref:          r.ExtensibleManagedObject.Reference(),
+			parentRef:    r.Parent,
+			customValues: e.loadCustomAttributes(&r.ManagedEntity),
+		}
+	}
+	return m, nil
+}
+
+func getResourcePoolName(ctx context.Context, e *Endpoint, rp types.ManagedObjectReference, rps objectMap) string {
```

Comment:
Looks like `context` and `e` aren't used in this function
```suggestion
func getResourcePoolName(rp types.ManagedObjectReference, rps objectMap) string {
```